thrift/thrift0.13/src/test/java/com/linecorp/armeria/server/thrift/THttpServiceBlockingTest.java

```diff
@@ -0,0 +1,127 @@
+
+    private static final String BLOCKING_EXECUTOR_PREFIX = "blocking-test";
+    private static final ScheduledExecutorService executor =
+            new ScheduledThreadPoolExecutor(1,
+                                            ThreadFactories.newThreadFactory(BLOCKING_EXECUTOR_PREFIX, true));
+
+    @BeforeEach
+    void clearDetector() {
+        blocking.set(false);
+    }
+
+    @AfterAll
+    public static void shutdownExecutor() {
+        executor.shutdown();
+    }
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+
+            sb.service("/",
+                       THttpService.builder().addService(new HelloServiceAsyncImpl()).build());
+            sb.service("/blocking", THttpService.builder()
+                                                .useBlockingTaskExecutor(true)
+                                                .addService(new HelloServiceAsyncImpl())
+                                                .build());
+            sb.service("/blocking-iface",
+                       THttpService.builder().addService(new HelloServiceImpl()).build());
+
+            sb.blockingTaskExecutor(executor, true);
+        }
+    };
+
+    @Test
+    void nonBlocking() throws Exception {
+        final HelloService.Iface client = ThriftClients.newClient(server.httpUri(), HelloService.Iface.class);
+
+        final String message = "nonBlockingTest";
+        final String response = client.hello(message);
+
+        assertThat(response).isEqualTo(message);
+        assertThat(blocking).isFalse();
+    }
+
+    @Test
+    void blocking() throws Exception {
+        final HelloService.Iface client =
+                ThriftClients.builder(server.httpUri())
+                             .path("/blocking")
+                             .build(HelloService.Iface.class);
+        final String message = "blockingTest";
+        final String response = client.hello(message);
+
+        assertThat(response).isEqualTo(message);
+        assertThat(blocking).isTrue();
+    }
+
+    @Test
+    void blockingIface() throws Exception {
+        final HelloService.Iface client =
+                ThriftClients.builder(server.httpUri())
+                             .path("/blocking-iface")
+                             .build(HelloService.Iface.class);
+        final String message = "blockingTest";
+        final String response = client.hello(message);
+
+        assertThat(response).isEqualTo(message);
+        assertThat(blocking).isTrue();
+    }
+
+    static class HelloServiceAsyncImpl implements HelloService.AsyncIface {
+        @Override
+        public void hello(String name, AsyncMethodCallback resultHandler) throws TException {
+            resultHandler.onComplete(name);
+            blocking.set(Thread.currentThread().getName().startsWith(BLOCKING_EXECUTOR_PREFIX));
```

Comment:
nit; swapping these two may help with flakiness. Otherwise, the response may be sent back and the main test thread may proceed before the blocking value is set

```suggestion
            blocking.set(Thread.currentThread().getName().startsWith(BLOCKING_EXECUTOR_PREFIX));
            resultHandler.onComplete(name);
```

Reply:
@jrhee17 Thank you for your comment. I applied it.
By the way, how can I know which test is failed from failed build?
I cannot find the failed tests from CI/CD

Reply:
@ChangguHan You can check the build scan reports here: https://github.com/line/armeria/pull/5619#issuecomment-2063069104 which are updated when builds are complete. Open the report and go to the 'Tests' section to browse the test result report.

Reply:
For example, there's only one failed job and it failed due to a flaky test which seems irrelevant to your work, so it's all good! :green_circle: https://ge.armeria.dev/s/yq5kpsjeqryju/tests/overview